onnxruntime/test/python/transformers/test_whisper.py

```diff
@@ -0,0 +1,115 @@
+# license information.
+# --------------------------------------------------------------------------
+
+import os
+import unittest
+
+import onnx
+from whisper_model_generator import *
+from parity_utilities import find_transformers_source
+
+if find_transformers_source():
+    from fusion_options import FusionOptions
+    from onnx_model import OnnxModel
+    from optimizer import optimize_model
+else:
+    from onnxruntime.transformers.fusion_options import FusionOptions
+    from onnxruntime.transformers.onnx_model import OnnxModel
+    from onnxruntime.transformers.optimizer import optimize_model
+
+
+class TestFusion(unittest.TestCase):
+    def verify_fusion(self, optimized_model, expected_model_filename):
+        optimized_model.topological_sort(is_deterministic=True)
+
+        expected_model_path = os.path.join(os.path.dirname(__file__), "test_data", "whisper", expected_model_filename)
+        expected_model = OnnxModel(onnx.load(expected_model_path))
+        expected_model.topological_sort(is_deterministic=True)
+
+        if "sln" in expected_model_filename or "mha" in expected_model_filename:
+            with open("optimized_model.txt", "w") as f:
+                f.write(str(optimized_model.model.graph))
+                f.close()
+            with open("expected_model.txt", "w") as f:
+                f.write(str(expected_model.model.graph))
+                f.close()
+        self.assertEqual(str(optimized_model.model.graph), str(expected_model.model.graph))
+
+    # Attention type #1 in onnx_model_bart.py
+    def test_encoder_attention_fusion_with_skiplayernorm(self):
+        num_heads = 4
+        hidden_size = 64
+        model = create_whisper_encoder_attention(num_heads=num_heads, hidden_size=hidden_size, add_before_layernorm=False)
+        dir = "."
+        model_path = os.path.join(dir, "whisper_encoder_attention_sln.onnx")
+        onnx.save(model, model_path)
+        options = FusionOptions("bart")
+        optimized_model = optimize_model(model_path, model_type="bart", num_heads=num_heads, hidden_size=hidden_size, 
+                                         optimization_options=options)
+        os.remove(model_path)
+        self.verify_fusion(optimized_model, "encoder_attention_with_sln_fused.onnx")
+
+    # Attention type #2 in onnx_model_bart.py
+    def test_decoder_attention_fusion_with_skiplayernorm(self):
+        num_heads = 4
+        hidden_size = 64
+        model = create_whisper_decoder_attention(num_heads=num_heads, hidden_size=hidden_size, add_before_layernorm=False)
+        dir = "."
+        model_path = os.path.join(dir, "whisper_decoder_attention_sln.onnx")
+        onnx.save(model, model_path)
+        options = FusionOptions("bart")
+        optimized_model = optimize_model(model_path, model_type="bart", num_heads=num_heads, hidden_size=hidden_size, 
+                                         optimization_options=options)
+        os.remove(model_path)
```

Comment:
Consider using tempfile for this https://docs.python.org/3/library/tempfile.html